File: Advanced/GuiDemo/app/src/main/java/edu/cs4730/guidemo/Relativelayout_Fragment.java
package edu.cs4730.guidemo;

import android.app.Activity;
import android.content.Context;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;


public class Relativelayout_Fragment extends Fragment {
	
	
    String TAG = "Relativelayout_Fragment";
    Context myContext;

	public Relativelayout_Fragment() {
		// Required empty public constructor
	}

	@Override
	public void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);

	}
	@Override
	public View onCreateView( LayoutInflater inflater, ViewGroup container,	Bundle savedInstanceState )	{
		View myView;
		myView = inflater.inflate( R.layout.relativelayout_fragment, container, false );
		return myView;
	}
	@Override
	public void onAttach(Activity activity) {
		super.onAttach(activity);
		myContext = activity.getApplicationContext();
		Log.d(TAG,"onAttach");
	}

	

}

File: Advanced/GuiDemo_kt/app/src/main/java/edu/cs4730/guidemo_kt/Relativelayout_Fragment.kt
package edu.cs4730.guidemo_kt

import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.content.Context
import android.util.Log
import android.view.View
import androidx.fragment.app.Fragment

class Relativelayout_Fragment : Fragment() {
    var TAG = "Relativelayout_Fragment"
    lateinit var myContext: Context
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val myView: View
        myView = inflater.inflate(R.layout.relativelayout_fragment, container, false)
        return myView
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        myContext = context
        Log.d(TAG, "onAttach")
    }
}